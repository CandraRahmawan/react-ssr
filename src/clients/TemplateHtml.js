import React, {Component} from 'react';
import Entry from './Entry';

export default class TemplateHtml extends Component {
  render() {
    return (
      <html>
      <head>
        <title>Server Side Rendering with ReactJS</title>
      </head>
      <body>
      <div id="app"><Entry/></div>
      </body>
      </html>
    )
  }
}