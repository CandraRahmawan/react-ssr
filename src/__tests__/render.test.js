import TemplateHtml from '../clients/TemplateHtml';

test('test component', () => {
  const component = renderer.create(
    <TemplateHtml entry="testing bro...."/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});