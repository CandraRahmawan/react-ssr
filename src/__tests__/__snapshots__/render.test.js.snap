// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component 1`] = `
<html>
  <head>
    <title>
      Server Side Rendering with ReactJS
    </title>
  </head>
  <body>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "testing bro....",
        }
      }
      id="app"
    />
  </body>
  <script
    src="main.bundle.js"
  />
</html>
`;
